packages/runtime/container-runtime/src/gc/garbageCollection.ts

```diff
@@ -990,7 +1000,9 @@ export class GarbageCollector implements IGarbageCollector {
 		request,
 		headerData,
 	}: IGCNodeUpdatedProps) {
-		if (!this.shouldRunGC) {
+		// If there is no reference timestamp to work with, no ops have been processed after creation. If so, skip
+		// logging as nothing interesting would have happened worth logging.
+		if (!this.shouldRunGC || timestampMs === undefined) {
```

Comment:
You can remove the `timestampMs ?? this.runtime.getCurrentReferenceTimestampMs()` below.

But I was actually coming here to suggest the opposite - why have all the callers have to think about getCurrentReferenceTimestampMs(), rather than just using undefined and doing it here?  I'll go re-read the PR descriptions to see if you explained more that I forgot.

Reply:
Except loading, everything else should be based on ops and the callers should provide that for precision. GC could call `getCurrentReferenceTimestampMs()` but the timestamp could be different than the op for which GC is called. Plus it's a small perf improvement to not having to call the function when the timestamp is alreayd present at the call site.

To summarize, for loads, we could call `getCurrentReferenceTimestampMs()` and for others callers should pass in the timestamp. For consistency and simplicity, the caller could always provide it.

Reply:
k